pkg/liqo-controller-manager/offloadingStatus-controller/offloadingStatus_controller.go

```diff
@@ -0,0 +1,135 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package offloadingstatuscontroller
+
+import (
+	"context"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// OffloadingStatusReconciler checks the status of all remote namespaces associated with this
+// namespaceOffloading Resource, and sets the global offloading status only when there is a feedback
+// from all remote clusters.
+type OffloadingStatusReconciler struct {
+	client.Client
+	Scheme               *runtime.Scheme
+	RemoteClustersNumber int
+}
+
+// Reconcile checks if all remote Namespaces have been created with success or not.
+func (r *OffloadingStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      liqoconst.DefaultNameNamespaceOffloading,
+	}, namespaceOffloading); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no NamespaceOffloading resource in Namespace '%s'", req.Namespace)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("%s --> Unable to get namespaceOffloading for the namespace '%s'", err, req.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	if r.RemoteClustersNumber == 0 {
```

Comment:
the number of the remote clusters may change over time, here we are only setting it at the start time

Reply:
yes at the moment this is the intended behavior, we consider as "valid for offloading" only nodes that exist at NamespaceOffloading CR creation; so i have to check only these nodes in order to decide the final state.

Reply:
ok, I didn't know it, thanks 👍